tests/test_filter_jobs.py

```diff
@@ -11,10 +11,16 @@
 def test_filter_jobs(base_url, selenium):
     """Open resultset page and filter for platform"""
     page = TreeherderPage(selenium, base_url).open()
-    platform = u'Linux'
+    all_builds = list(page.all_builds)
 
+    platform = u'Linux'
     page.filter_by(platform)
-    assert platform in page.result_sets[0].builds[0].platform_name
+    filtered_builds = list(page.all_builds)
+
+    buildname = page.result_sets[0].builds[0].platform_name
+    assert platform in buildname
+
+    assert not all_builds == filtered_builds
```

Comment:
There's a lot that could go wrong here and this would still pass. Perhaps we could also compare the lengths of these lists (filtered should be less), or maybe even check that filtered builds is a subset of all builds.